Hospital/HospitalApi/Controllers/RoomController.cs

```diff
@@ -44,5 +44,17 @@ public Room UpdateRoom(Room room)
             var roomRepo = _uow.GetRepository<IRoomWriteRepository>();
             return roomRepo.Update(room);
         }
+
+
+        [HttpGet("find")]      
+        public IActionResult FindByName([FromQuery(Name = "roomName")] string roomName, [FromQuery(Name = "buildingName")] string buildingName)
```

Comment:
Rename method to "FindByNameAndBuildingName".
Since you're inside RoomController you can use just "name" instead of "roomName", we assume that "name" reffers to Room. (Just like inside Room class in model you have Name and BuildingName props)